change `authorization_details` to use `credential_description` from metadata (#137)
openid/OpenID4VCI: openid-4-verifiable-credential-issuance-1_0.md

The request parameter `authorization_details` defined in Section 2 of [@!RFC9396] MUST be used to convey the details about the Credentials the Wallet wants to obtain. This specification introduces a new authorization details type `openid_credential` and defines the following parameters to be used with this authorization details type:

* `type`: REQUIRED. String that determines the authorization details type. MUST be set to `openid_credential` for the purpose of this specification.

> * `credential_description`: REQUIRED. String specifying the name of a Credential described in the `credentials_supported` from the Credential Issuer Metadata as described in (#credential-issuer-parameters). The referenced metadata object conveys the details, e.g. format, for the requested Credential issuance. This specification defines Credential Format specific Issuer Metadata in (#format_profiles).

Reviewer: ```suggestion
* `credential_description`: REQUIRED. String specifying a unique identifier of the Credential being described in the `credentials_supported` map in the Credential Issuer Metadata as defined in (#credential-issuer-parameters). The referenced object in the `credentials_supported` map conveys the details, e.g. format, for the requested Credential issuance. This specification defines Credential Format specific Issuer Metadata in (#format_profiles).
```

> * `credential_description`: REQUIRED. String specifying the name of a Credential described in the `credentials_supported` from the Credential Issuer Metadata as described in (#credential-issuer-parameters). The referenced metadata object conveys the details, e.g. format, for the requested Credential issuance. This specification defines Credential Format specific Issuer Metadata in (#format_profiles).

Reviewer: I am also unsure about the `credential_description` parameter name. in Credential Offer, we simply have a `credentials` array. would it be sufficient to say `credential` here?

> * `credential`: REQUIRED. String specifying a unique identifier of the Credential being described in the `credentials_supported` map in the Credential Issuer Metadata as defined in (#credential-issuer-parameters). The referenced object in the `credentials_supported` map conveys the details, e.g. format, for the requested Credential issuance. This specification defines Credential Format specific Issuer Metadata in (#format_profiles).

Reviewer 2: ```suggestion
* `credential_identifier`: REQUIRED. String specifying a unique identifier of the Credential being described in the `credentials_supported` map in the Credential Issuer Metadata as defined in (#credential-issuer-parameters). The referenced object in the `credentials_supported` map conveys the details, e.g. format, for the requested Credential issuance. This specification defines Credential Format specific Issuer Metadata in (#format_profiles).
```

Reviewer 2: Calling something `credential` and it is not a credential strikes me.

Reviewer 3: >  Calling something `credential` and it is not a credential strikes me.

Concur with the sentiment. But `credential_identifier` is (or will be) used elsewhere, no?

Reviewer 4: The problem is that we don't have a clear concept here and therefore no clear name. We discussed this on the last call and agreed that we need to fix this in Issue #91.

Author: I named it `credential_description` beforehand, argument was that credential_offer references the same thing and it's named `credentials` there. `credential_identifier` is used in Token Response/Credential Request.

Reviewer: if we can agree on something like `credential_configuration`, I am ok. but it cannot be `credential_identifier`.

Reviewer: discussed on DCP WG call, agreed on `credentials_supported_id` for now. might change if `credentials_supported` parameter itself gets renamed...

  &client_id=s6BhdRkqt3
  &code_challenge=E9Melhoa2OwvFrEMTJguCHaoeK1t8URWbuGJSstw-cM
  &code_challenge_method=S256

>   &authorization_details=%5B%7B%22type%22:%20%22openid_credential%22%0A%20%20%20%20,
>     %20%22credentials_supported_id%22:%20%22UniversityDegreeCredential%22%7D%5D

Reviewer 3: this doesn't look quite right - the comma and colon shouldn't be in there?

```suggestion
  &authorization_details=%5B%7B%22type%22%3A%20%22openid_credential%22%2C%20%22
    credentials_supported_id%22%3A%20%22UniversityDegreeCredential%22%7D%5D
```

* `c_nonce`: OPTIONAL. String containing a nonce to be used when creating a proof of possession of the key proof (see (#credential_request)). When received, the Wallet MUST use this nonce value for its subsequent requests until the Credential Issuer provides a fresh nonce.
* `c_nonce_expires_in`: OPTIONAL. Number denoting the lifetime in seconds of the `c_nonce`.

> * `authorization_details`: REQUIRED when `authorization_details` parameter is used to request issuance of a certain Credential type as defined in (#authorization-details). MUST NOT be used otherwise. An array of objects as defined in Section 7 of [@!RFC9396]. This specification defines the following parameter to be used with authorization details type `openid_credential` in the Token Response:
>   * `credential_description`: REQUIRED. String specifying the name of a Credential described in the `credentials_supported` from the Credential Issuer Metadata as described in (#credential-issuer-parameters). The referenced metadata object conveys the details, e.g. format, for the requested Credential issuance. This specification defines Credential Format specific Issuer Metadata in (#format_profiles).

Reviewer: ```suggestion
* `authorization_details`: REQUIRED when `authorization_details` parameter is used to request issuance of a certain Credential type as defined in (#authorization-details). MUST NOT be used otherwise. An array of objects as defined in Section 7 of [@!RFC9396]. In addition to the new parameter defined in (#authorization-details), this specification defines the following parameter to be used with authorization details type `openid_credential` in the Token Response:
```
we should either redefine both `type` and `credential_description` parameters here, or simply point to the previous section..
